### PR TITLE
add comments to jshint rules

### PR DESCRIPTION
**Pull request by thanpolas**
This PR adds sensible comments to the `.jshintrc` file, retains every rule that previously existed and adds the following new rules:

```
  // Force all variable names to use either camelCase style
  // or UPPER_CASE with underscores.
  "camelcase": true,
  // Prohibit trailing whitespace.
  "trailing": true,
```

Additionally adds the following two opinionated rules, but under a clear banner to easily be configured. Opinions definitely diverge here but the rules have to be there for developers to change.

```
  // Enforce tab width of 2 spaces.
  "indent": 2,
  // Enforce use of single quotation marks for strings.
  "quotmark": "single",
```

Finally, two more rules were added related to not warn for the lack of `"use strict"`. I understand if you do not wish this be included.

```
  // do not complain for lack of 'use strict';
  "globalstrict": false,
  "strict": false,
```

### CODE REVIEW

File: app/templates/jshintrc [hunk 1]
```diff
@@ -1,23 +1,107 @@
...
+
+  // Define globals exposed by Node.js.
+  "node": true,
+
+  /*
+   * TABS AND QUOTES
+   * ==================
+   */
+  // Enforce tab width of 2 spaces.
+  "indent": 2,
```
Comment (MrDHat) [inline app/templates/jshintrc]: Enforcing this might not be a good idea.

File: app/templates/jshintrc [hunk 2]
```diff
@@ -1,23 +1,107 @@
...
+
+  // Enforce use of single quotation marks for strings.
+  "quotmark": "single",
```
Comment (MrDHat) [inline app/templates/jshintrc]: Enforcing this might not be a good idea.

File: app/templates/jshintrc (hunk 1 above)
Comment (thanpolas) [inline app/templates/jshintrc]: > Opinions definitely diverge here but the rules have to be there for developers to change.

prompt user or burn?

File: app/templates/jshintrc [hunk 3]
```diff
@@ -1,23 +1,107 @@
...
+  // deprecated in future versions of JavaScript.
+  "noarg": true,
+
+  // This option suppresses warnings about the use of
+  // assignments in cases where comparisons are expected
+  "boss": true,
+
+  // Force all variable names to use either camelCase style
+  // or UPPER_CASE with underscores.
+  "camelcase": true,
```
Comment (MrDHat) [inline app/templates/jshintrc]: Enforcing might not be a good idea.

File: app/templates/jshintrc (hunk 1 above)
Comment (MrDHat) [inline app/templates/jshintrc]: I have tried to keep these in line with the official generators. See: https://github.com/yeoman/generator-node/blob/master/app/templates/jshintrc
I think we should remove this rule so as to maintain uniformity. Same goes for other similar options.

File: app/templates/jshintrc [hunk 4]
```diff
@@ -1,23 +1,107 @@
...
-      "test": false,
-      "before": false,
-      "beforeEach": false,
-      "after": false,
-      "afterEach": false
-    }
+
+  // do not complain for lack of 'use strict';
+  "globalstrict": false,
+  "strict": false,
```
Comment (MrDHat) [inline app/templates/jshintrc]: Lets leave this for the users to decide. For the sake of uniformity :)

File: app/templates/jshintrc (hunk 1 above)
Comment (thanpolas) [inline app/templates/jshintrc]: you mean remove `camelcase` and `trailing` too?

Comment (MrDHat) [inline app/templates/jshintrc]: `trailing` makes sense. Please remove `indent`, `camelCase`, `quotmark`, `globalstrict` and `strict`.

**Comment by thanpolas** [comment]
@MrDHat changes incorporated, ready.

File: app/templates/jshintrc [hunk 5]
```diff
@@ -1,23 +1,89 @@
+/**
+ * Project specific jshint linting options.
+ *
+ * Forked from Airbnb JSHint settings.
+ *
+ * @version 0.4.0s
```
Comment (MrDHat) [inline app/templates/jshintrc:6]: What does this represent?

Comment (thanpolas) [inline app/templates/jshintrc:6]: jshint evolves over time, [i treat jshint as a stand-alone entity](https://github.com/thanpolas/javascript/blob/master/linters/jshintrc), thus the versioning.

**Comment by MrDHat** [comment]
This looks good now. Please squash the last three commits into one and I'll be happy to merge :smile:

**Comment by thanpolas** [comment]
I am sorry, i can't do that.

**Comment by MrDHat** [comment]
Erm.. I asked you to do that, so that only those commits appear which are necessary. Nevertheless, I understand. Merging now.
Thank you for the patch!

**Pull request merged**